### PR TITLE
Split Iceberg cloud tests into a separate CI job

### PR DESCRIPTION
**Pull request by alexjo2144**
## Description

The Iceberg tests are coming close to hitting the 60 minute timeout. This splits the cloud tests into a separate job to reduce the total time.

## Non-technical explanation

Make tests run faster.

## Release notes

(x) This is not user-visible and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -534,7 +538,9 @@ jobs:
           AWS_REGION: us-east-2
           S3_BUCKET: presto-ci-test
           GCP_CREDENTIALS_KEY: ${{ secrets.GCP_CREDENTIALS_KEY }}
-        if: contains(matrix.modules, 'plugin/trino-iceberg') && (env.AWS_ACCESS_KEY_ID != '' || env.AWS_SECRET_ACCESS_KEY != '' || env.GCP_CREDENTIALS_KEY != '')
+        if: >-
+          contains(matrix.modules, 'trino-iceberg') && contains(matrix.profile, 'cloud-tests') &&
+          (env.AWS_ACCESS_KEY_ID != '' || env.AWS_SECRET_ACCESS_KEY != '' || env.GCP_CREDENTIALS_KEY != '')
         run: |
           $MAVEN test ${MAVEN_TEST} -pl :trino-iceberg -P cloud-tests \
```
Comment (findepi) [inline .github/workflows/ci.yml:545]: Should this be `format('-P {0}', matrix.profile)` now?

Comment (alexjo2144) [reply .github/workflows/ci.yml]: Sure, updated

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -494,7 +498,7 @@ jobs:
           contains(matrix.modules, 'trino-delta-lake') && contains(matrix.profile, 'cloud-tests') &&
           (env.ABFS_ACCOUNT != '' || env.ABFS_CONTAINER != '' || env.ABFS_ACCESSKEY != '' || env.AWS_ACCESS_KEY_ID != '' || env.AWS_SECRET_ACCESS_KEY != '')
         run: |
-          $MAVEN test ${MAVEN_TEST} -P cloud-tests -pl :trino-delta-lake \
+          $MAVEN test ${MAVEN_TEST} format('-P {0}', matrix.profile) -pl :trino-delta-lake \
```
Comment (ebyhr) [inline .github/workflows/ci.yml:501]: `${{ }}` is missing. https://github.com/trinodb/trino/pull/14087/commits/b319b127bfa00a405e6053591bddcb20e9d23462

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -477,7 +478,10 @@ jobs:
           export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
           $RETRY $MAVEN clean install ${MAVEN_FAST_INSTALL} ${MAVEN_GIB} -am -pl "${{ matrix.modules }}"
       - name: Maven Tests
-        if: matrix.modules != 'plugin/trino-singlestore' && !(contains(matrix.modules, 'trino-delta-lake') && contains(matrix.profile, 'cloud-tests'))
+        if: >-
+          matrix.modules != 'plugin/trino-singlestore' &&
+          !(contains(matrix.modules, 'trino-delta-lake') && contains(matrix.profile, 'cloud-tests')) &&
+          !(contains(matrix.modules, 'trino-iceberg') && contains(matrix.profile, 'cloud-tests'))
```
Comment (nineinchnick) [inline .github/workflows/ci.yml:484]: The `&& contains(matrix.profile, 'cloud-tests'))` condition is duplicated.

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**